sdk/tests/conformance2/extensions/ext-color-buffer-float.html

```diff
@@ -334,6 +334,116 @@
+            destInternalformat: "RGB16F",
+            data: new Uint16Array(width * height * 4)
+        },
+        {
+            internalformat: "RGBA32F",
+            format: "RGBA",
+            type: "FLOAT",
+            destInternalformat: "RGB32F",
+            data: new Float32Array(width * height * 4)
+        },
+        {
+            internalformat: "RGBA16F",
+            format: "RGBA",
+            type: "HALF_FLOAT",
+            destInternalformat: "RGBA16F",
+            data: new Uint16Array(width * height * 4)
+        },
+        {
+            internalformat: "RGBA32F",
+            format: "RGBA",
+            type: "FLOAT",
+            destInternalformat: "RGBA32F",
+            data: new Float32Array(width * height * 4)
+        },
+        {
+            internalformat: "R11F_G11F_B10F",
+            format: "RGB",
+            type: "FLOAT",
+            destInternalformat: "R11F_G11F_B10F",
+            data: new Float32Array(width * height * 3)
+        }
+    ];
+    cases.forEach(function(testcase) {
+        debug("");
+        debug("Testing CopyTexImage2D for internalformat: " + testcase.destInternalformat);
+
+        var internalformat = gl[testcase.internalformat];
+        var format = gl[testcase.format];
+        var type = gl[testcase.type];
+        var destInternalformat = gl[testcase.destInternalformat];
+        var texSrc = gl.createTexture();
+        gl.bindTexture(gl.TEXTURE_2D, texSrc);
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+        var data = testcase.data;
+        gl.texImage2D(gl.TEXTURE_2D, level, internalformat, width, height, 0, format, type, data);
+        var fbo = gl.createFramebuffer();
+        gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+        gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texSrc, level);
+        var texDest = gl.createTexture();
+        gl.bindTexture(gl.TEXTURE_2D, texDest);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Setup framebuffer with texture should succeed.");
+        if (enabled) {
+            shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)", "gl.FRAMEBUFFER_COMPLETE");
+            gl.copyTexImage2D(gl.TEXTURE_2D, level, destInternalformat, 0, 0, width, height, 0);
+            if (destInternalformat == gl.RGB16F || destInternalformat == gl.RGB32F) {
+                wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "CopyTexImage2D should fail.");
+            } else {
+                wtu.glErrorShouldBe(gl, gl.NO_ERROR, "CopyTexImage2D should succeed.");
+            }
+        } else {
+            shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)", "gl.FRAMEBUFFER_INCOMPLETE_ATTACHMENT");
+            gl.copyTexImage2D(gl.TEXTURE_2D, level, destInternalformat, 0, 0, width, height, 0);
+            wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "CopyTexImage2D should fail.");
```

Comment:
This needs to be more flexible, because there are multiple error cases here, for example, the fbo isn't complete, that should trigger an error other than INVALID_ENUM.